cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -153,20 +153,15 @@ std::unique_ptr<nav_msgs::OccupancyGrid>
 MapBuilderBridge::BuildOccupancyGrid() {
   CHECK(node_options_.map_builder_options.use_trajectory_builder_2d())
       << "Publishing OccupancyGrids for 3D data is not yet supported";
-  std::vector<::cartographer::mapping::TrajectoryNode> flat_trajectory_nodes;
-  for (const auto& single_trajectory_nodes :
-       map_builder_.sparse_pose_graph()->GetTrajectoryNodes()) {
-    flat_trajectory_nodes.insert(flat_trajectory_nodes.end(),
-                                 single_trajectory_nodes.begin(),
-                                 single_trajectory_nodes.end());
-  }
+  const auto all_trajectory_nodes =
+      map_builder_.sparse_pose_graph()->GetTrajectoryNodes();
   std::unique_ptr<nav_msgs::OccupancyGrid> occupancy_grid;
-  if (!flat_trajectory_nodes.empty()) {
+  if (!all_trajectory_nodes.empty()) {
```

Comment:
This should verify that there is any non-trimmed node, not that there is a trajectory?